pkg/azureclients/armclient/util.go

```diff
@@ -129,9 +129,10 @@ func DoHackRegionalRetryDecorator(c *Client) autorest.SendDecorator {
 			}
 			klog.V(5).Infof("Send.sendRequest original response: %s", bodyString)
 
-			if err, ok := body["error"].(map[string]interface{}); !ok ||
-				err["code"] == nil ||
-				!strings.EqualFold(err["code"].(string), "ResourceGroupNotFound") {
+			err, ok := body["error"].(map[string]interface{})
+			if !ok || err["code"] == nil ||
+				(!strings.EqualFold(err["code"].(string), "ResourceGroupNotFound") &&
+					(response.ContentLength != 0 || response.StatusCode != http.StatusOK)) {
```

Comment:
Minor concern: can we add comments here to improve readability? And personally, if there is a complex if condition, I will assign it a meaningful variable. For example, this piece would be like:

```go
// Need to call the regional host because ...., check xxxx for details.
emptyResp := response.ContentLenth == 0 && response.StatusCode >= 200 && response.StatusCode < 300
if !ok || err["code"] == nil || !strings.EqualFold(err["code"].(string), "ResourceGroupNotFound" || !emptyResp {
  ...
}
```
But I won't block this one because the readability is not the key.

Reply:
Makes sense. Updated.